packages/vips/src/index.ts

```diff
@@ -28,18 +43,27 @@ type ItemId = string;
  *
  * Reuses any existing instance.
  */
-async function getVips(): Promise< typeof VipsInstance > {
+async function getVips(): Promise< typeof Vips > {
 	if ( vipsInstance ) {
 		return vipsInstance;
 	}
 
-	const mainBlobUrl = URL.createObjectURL(
-		await ( await fetch( `${ VIPS_CDN_URL }/vips.js` ) ).blob()
-	);
-
 	vipsInstance = await Vips( {
-		locateFile: ( fileName: string ) => `${ VIPS_CDN_URL }/${ fileName }`,
-		mainScriptUrlOrBlob: mainBlobUrl,
+		locateFile: ( fileName: string ) => {
+			if ( fileName.endsWith( 'vips.wasm' ) ) {
+				fileName = VipsModule;
+			} else if ( fileName.endsWith( 'vips-heif.wasm' ) ) {
+				fileName = VipsHeifModule;
+			} else if ( fileName.endsWith( 'vips-jxl.wasm' ) ) {
+				fileName = VipsJxlModule;
+			}
+
+			// fileName is a blob:<...> URL.
+			// Only get the path name and append it to the provided location.
+			return (
+				location + new URL( fileName.replace( 'blob:', '' ) ).pathname
```

Comment:
FWIW, this `blob:` URL can be avoided by setting `publicPath` to an empty string here:
https://github.com/swissspidy/media-experiments/blob/7f2c098521277978f48f38dcfe0f6528836e04b9/webpack.config.js#L106

Reply:
Ah, neat! That makes it a bit easier to read :-)